src/components/search/geolocation-button/index.tsx

```diff
@@ -2,16 +2,31 @@ import { Component, useEffect, useState } from 'react';
 import { LoadingIcon } from '@atb/components/loading';
 import { reverse } from '@atb/page-modules/departures/client';
 import { GeocoderFeature } from '@atb/page-modules/departures';
-import { ComponentText, useTranslation } from '@atb/translations';
+import {
+  ComponentText,
+  TranslateFunction,
+  useTranslation,
+} from '@atb/translations';
 import { MonoIcon } from '@atb/components/icon';
 
 type GeolocationButtonProps = {
   onGeolocate: (feature: GeocoderFeature) => void;
+  onError?: (error: string | null) => void;
   className?: string;
 };
-function GeolocationButton({ onGeolocate, className }: GeolocationButtonProps) {
+function GeolocationButton({
+  onGeolocate,
+  onError,
+  className,
+}: GeolocationButtonProps) {
   const { t } = useTranslation();
-  const { getPosition, isLoading, isUnavailable } = useGeolocation(onGeolocate);
+  const { getPosition, isLoading, isUnavailable, error } =
+    useGeolocation(onGeolocate);
+
+  useEffect(() => {
+    if (error) onError?.(getErrorMessage(error.code, t));
+    else onError?.(null);
```

Comment:
wonder if onError rather should be passed to useGeolocation instead of having to use effect here. So just use it as a normal callback

Reply:
Good idea, implemented that.